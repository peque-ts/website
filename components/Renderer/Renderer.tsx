import React from 'react';

import { useCode } from '../../hooks/use-code';

interface Props {
  html: string;
}

export const Renderer: React.VFC<Props> = ({ html }) => {
  useCode();

  return <article className="renderer" dangerouslySetInnerHTML={{ __html: html }} />;
};
