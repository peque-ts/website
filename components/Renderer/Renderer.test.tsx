import { render, screen } from '@testing-library/react';

import { Renderer } from './Renderer';

describe('Renderer', () => {
  it('should render', () => {
    const { asFragment } = render(<Renderer html={'<p data-testid="paragraph">hello world</p>'} />);

    expect(screen.getByTestId('paragraph')).toHaveTextContent('hello world');
    expect(asFragment()).toMatchSnapshot();
  });
});
