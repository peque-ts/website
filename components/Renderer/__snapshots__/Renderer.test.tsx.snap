// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renderer should render 1`] = `
<DocumentFragment>
  <div
    class="renderer"
  >
    <p
      data-testid="paragraph"
    >
      hello world
    </p>
  </div>
</DocumentFragment>
`;
